### PR TITLE
highland: add missing definitions & correct some others

### PR DESCRIPTION
**Pull request by alvis**
What changed in this PR:
* added several missing function definitions, including `toPromise`, `uniq` and `uniqBy`
* corrected several definitions, including `each`, `merge` and `push`

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://highlandjs.org
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/highland/index.d.ts [hunk 1]
```diff
@@ -625,7 +627,7 @@ declare namespace Highland {
 		 * @param {Function} f - the function to pass all errors to
 		 * @api public
 		 */
-		errors(f: (err: Error, push: (err: Error, x?: R) => void) => void): Stream<R>;
+		  errors(f: (err: Error, push: (err: Error | null, x?: R) => void) => void): Stream<R>;
```
Comment (hgwood) [inline types/highland/index.d.ts]: Indentation looks incorrect here.

File: types/highland/index.d.ts [hunk 2]
```diff
@@ -71,8 +72,9 @@ interface HighlandStatic {
 	 * @api public
 	 */
 	<R>(): Highland.Stream<R>;
+  <R>(xs: Highland.Stream<R>[]): Highland.Stream<R>;
 	<R>(xs: R[]): Highland.Stream<R>;
-	<R>(xs: (push: (err: Error, x?: R) => void, next: () => void) => void): Highland.Stream<R>;
+	<R>(xs: (push: (err: Error | null, x?: R) => void, next: () => void) => void): Highland.Stream<R>;
```
Comment (hgwood) [inline types/highland/index.d.ts]: It might be a good idea to add tests for these cases. What do you think?

File: types/highland/index.d.ts [hunk 3]
```diff
@@ -1144,7 +1168,7 @@ declare namespace Highland {
 		 * @param {Function} f - the iterator function
 		 * @api public
 		 */
-		each(f: (x: R) => void): void;
+		each(f: (x: R) => void): Pick<Stream<R>, 'done'>;
```
Comment (hgwood) [inline types/highland/index.d.ts]: I can't find `Pick<T, U>`. Could you tell me where it is defined?

**Review by hgwood — COMMENTED** [review]
Thank you

File: types/highland/index.d.ts (hunk 3 above)
Comment (alvis) [reply types/highland/index.d.ts]: It's a new build-in type introduced in TS 2.1
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html

File: types/highland/index.d.ts (hunk 1 above)
Comment (alvis) [reply types/highland/index.d.ts]: Thanks for pointing out. :)

File: types/highland/index.d.ts (hunk 2 above)
Comment (alvis) [reply types/highland/index.d.ts]: This declaration is corresponding to `Async` in http://highlandjs.org/#async

In fact, it should have covered by the test here
https://github.com/alvis/DefinitelyTyped/blob/471ce58849ca7a17de1b96f504ec08fc3f7219fa/types/highland/highland-tests.ts#L129

Comment (hgwood) [reply types/highland/index.d.ts]: Right! Now I'm wondering why it was working before. Why did [`push` accept `null`](https://github.com/alvis/DefinitelyTyped/blob/471ce58849ca7a17de1b96f504ec08fc3f7219fa/types/highland/highland-tests.ts#L130) as a first argument before your change?

File: types/highland/index.d.ts (hunk 3 above)
Comment (hgwood) [reply types/highland/index.d.ts]: Oh, neat! Thanks.

File: types/highland/index.d.ts (hunk 2 above)
Comment (alvis) [reply types/highland/index.d.ts]: Interesting question. I wondered that too.

Comment (alvis) [reply types/highland/index.d.ts]: I think it worked because `strictNullChecks` was only introduced recently.

Comment (hgwood) [reply types/highland/index.d.ts]: I think you are right.

**Review by hgwood — APPROVED** [review]

**Pull request merged**